### PR TITLE
Adds BinaryName to tool and use that to look for the executable

### LINKED ISSUE #8: Regression testing with tools whose binary name differs from the project name
I was trying to use the utilities provided by this repo (downloading of releases and building binaries) to do some regression testing for Engine. The problem is that the `Name` of the `Tool` is currently used both for looking for the binary file and for building the name of the tar archive. In the case of Engine, the name of the binary is different from the prefix used for the tar archive.

So by setting the name of the `Tool` to `engine` I can download the asset from the release, but binary cannot be found and by setting the name of the `Tool` to `srcd` the binary would be found, but it exits earlier as it cannot found the asset. 

I've just started looking at the project (did I miss something?), but maybe the easiest thing would be to add a `BinaryName` field to the `Tool` struct?

### PR DESCRIPTION
**Pull request by se7entyse7en**
Closes #8.

### CODE REVIEW

File: config.go [hunk 1]
```diff
@@ -61,8 +61,10 @@ type BuildStep struct {
 
 // Tool describes a project to build and test.
 type Tool struct {
-	// Name has the tool name that is the same as the executable name.
+	// Name has the tool name.
 	Name string
+	// BinaryName has the executable name of the tool.
+	BinaryName string
```
Comment (smacker) [inline config.go:67]: does it make sense to keep backward compatibility and use `Name` if `BinaryName` is empty?

Comment (mcarmonaa) [reply config.go:67]: +1 to keep backward compatibility

Comment (se7entyse7en) [reply config.go:67]: Yeah it totally makes sense.

Comment (se7entyse7en) [reply config.go:67]: Addressed [here](https://github.com/src-d/regression-core/pull/9/commits/dd30bc1e7cb9026089c6c3ad78ef4606ea1691c8).

**Review by mcarmonaa — APPROVED** [review]

**Review by smacker — APPROVED** [review]
let's wait for Javi for the merge

**Review by jfontan — APPROVED** [review]

**Pull request merged**